DOC: Updated schema docs to prototype frame store

ericdill opened the pull request:
cc @danielballan @tacaswell @stuwilkins @arkilic

ericdill committed:
Merge branch 'setup' of https://github.com/ericdill/metadataStore into setup
MNT: local merge

ericdill committed:
DOC: Modified frame header template

Commit:
Merge pull request #1 from ericdill/setup

MNT: Added .gitignore and setup.py files

Commit:
Delete README.md

ericdill committed:
ENH: Added search keys dict

ericdill committed:
ENH: Added seed_broker.py to seed broker with an experiment from grad school

ericdill committed:
ENH: Added some complexity to include image uri's from xf23id1-ws3

ericdill committed:
MNT: Added print statement to host connections

ericdill committed:
ENH: Added search function to take a dict

Commit:
Merge pull request #2 from ericdill/vistrails_integration

Vistrails integration

Commit:
Schema and some dataapi changes. userapi changes to follow

Commit:
Added collectionapi and implemented some very raw functionality for type checks etc

Commit:
Added collectionapi and implemented some very raw functionality for type checks etc

Commit:
Delete README.md

Commit:
Create README.md

Commit:
Changes to schemas, raw_commands: insert and update default behavior changes, some find routine implementation via keywords
Future work: userapi api and dataCollection api changes

Commit:
Merge remote-tracking branch 'origin/orm_metadataStore' into orm_metadataStore

Commit:
Create README.md

Commit:
raw_commands and userapi commands header create added

Commit:
userapi.commands.record() added

Commit:
schema changes and dataapi.raw_commands.* modifications due to schema change

Commit:
sampleCode added

Commit:
data collection api modifications added

Commit:
collectionapi.commands: Header, EventDescriptor, and BeamlineConfig can be saved via create() routine

Commit:
data collection api modifications added, code cleaned from some debugging garbage

Commit:
metadataStore.collectionapi.commands.save() added

Commit:
data collection api record implementation and python docs. BeamlineConfig now returned as a part of search() via raw_commands. raw_commands.search(..., data=True) now returns events nested within their corresponding EventDescriptors

Commit:
Header indexing on return fixed, collectionaoi.search() added

Commit:
userapi.commands updated for the new schema. Sample code added for all apis

ericdill committed:
Added .gitignore file

ericdill committed:
MNT: Removed Arman's .idea files

Commit:
sample code added

Commit:
Merge branch 'orm_metadataStore' of https://github.com/ericdill/metadataStore into orm_metadataStore

ericdill committed:
MNT: Fixed bugs

ericdill committed:
Merge remote-tracking branch 'canon/orm_metadataStore' into orm_metadataStore

Conflicts:
	dataBroker.conf
	sample_code_userapi.py

Commit:
Merge branch 'orm_metadataStore' of https://github.com/ericdill/metadataStore into orm_metadataStore

ericdill committed:
MNT: Reverting changes

ericdill committed:
MNT: Removed local copy of sample_code_userapi.py

ericdill committed:
Merge pull request #6 from ericdill/orm_metadataStore

Orm metadata store

ericdill committed:
ENH: Added setup.py file

ericdill committed:
ENH: Added keyword dictionary and type checking to search

ericdill committed:
MNT: Minor changes to demo scripts (print statements, mostly)

Commit:
Merge pull request #7 from ericdill/orm_metadataStore

Orm metadata store

Commit:
Fixed descriptor_name vs. event_type_name ambiguity. Added type verification to scan_id for both create() and search()

Commit:
Merge remote-tracking branch 'origin/orm_metadataStore' into orm_metadataStore

Conflicts:
	sample_code_userapi.py

Commit:
Fixed descriptor_name vs. event_type_name ambiguity. Added type verification to scan_id for both create() and search()

ericdill committed:
MNT: Documented search and added type checking to input parameters

ericdill committed:
MNT: Removed space before colon in dict

ericdill committed:
MNT: Fixed docstring in search to comply with numpy

ericdill committed:
MNT: Added a todo

Commit:
Merge pull request #8 from ericdill/devel

MNT: Documented search and added type checking to input parameters

Commit:
Removed mongoengine: Change database tables and raw commands to make code dependent only on pymongo. Tested using the sample code for all supported apis

Commit:
Added unit test for databaseTable.Header

Commit:
Moved sample code for various apis into metadataStore.example. Added debian server start script to metadataStore.demo

ericdill committed:
MNT: Added checking for scan_id being None to commands.py

ericdill committed:
MNT: Raise a ValueError on bad search keys, don't log it silently

Commit:
Fixed no event return in search(.., data=True)

ericdill committed:
ENH: Added type checking and input validation to userapi.commands.search

ericdill committed:
Merge branch 'type_checking' into devel. This fixes #9

Commit:
Merge pull request #10 from ericdill/devel

Devel

Commit:
Added num_header field to search() in order to empower users to select the number of headers to be returned

Commit:
Merge remote-tracking branch 'origin/devel' into devel

Conflicts:
	metadataStore/userapi/commands.py

Commit:
Modified frame_hdr_events.txt to fit to latest schema. added todo items

Commit:
Documentation updated

Commit:
userapi is now using the databaseLogger instance like the rest of the code

ericdill committed:
MNT: Adding type-checking again

ericdill committed:
MNT: Fixed bugs in type/value validation in search()

Commit:
Added a common 'event_descriptors' key to headers that contain event_descriptor_# dicts

Commit:
Empty search() returns None

Commit:
Delete dataBroker.conf.orig

Commit:
Delete sample_code_userapi.py.orig

Commit:
Added edill as author to userapi. Commit&Push test via eclipse pydev

Commit:
name change on user api log to record and started a benchmark module for development

Commit:
Fixed

Commit:
Fixed behavior if config file does not exist and added a simple javascript to conveniently drop all tables(for developer branch and eyes only!!!)

Commit:
Added some benchmark code for insert speed tests

Commit:
Delete Outcome of all create run headers.png

ericdill committed:
Merge remote-tracking branch 'canon/devel' into devel

ericdill committed:
MNT: Fixed type checking

ericdill committed:
ENH: Added function stub for listify

ericdill committed:
ENH: Fleshed out the listify function a bit and added anothe helper function

ericdill committed:
BUG: Fixed behavior of listify

Commit:
Latest benchmarking code with graph and graphical profiling generators

Commit:
Merge pull request #16 from ericdill/devel

MNT: Fixed bugs in type/value validation in search()

Commit:
ENH: Copy dataBroker.conf to user home directory when setup.py is run

Commit:
get_data_keys() added to metadataStore.analysisapi.utility. Given a header, data keys are returned

ericdill committed:
MNT: Fixed get_data_keys to return actual data keys

Commit:
Event field validity check added to metadataStore.analysis.utility.get_data_keys()

ericdill committed:
Updated frame_hdr_events.txt to reflect current state of run header

ericdill committed:
Updated frame_hdr_events.txt to reflect current state of run header

ericdill committed:
Merge remote-tracking branch 'canon/devel' into devel

ericdill committed:
ENH: Added proposal for run header formatting

ericdill committed:
ENH: listify parses the current iteration of the run header correctly

ericdill committed:
MNT: Updated the proposed frame header

ericdill committed:
Updated frame_hdr_events.txt

ericdill committed:
ENH: Added _id to the search_keys_dict

ericdill committed:
MNT: Forced return val of get_data_keys to match return type specified in docstring

ericdill committed:
MNT: Removed default time behavior from listify

The default time behavior was to have time simply be a running
index because I previously did not include time in the event data
dict in the run header

ericdill committed:
MNT: Fixed print statements of demo behavior

Commit:
Bulk header create functionality added to collectionapi

Commit:
Merge remote-tracking branch 'origin/devel' into devel

Commit:
Prettier bulk header create functionality added to collectionapi. Now bulk insert is part of raw_commands rather than solely collectionapi

Commit:
Bulk event insert added to collection api and raw_commands

Commit:
Bulk insert features displayed in sample_collection_code. Massive performance improvement on inserts

ericdill committed:
MNT: Simplified listify

ericdill committed:
changes to proposed frame header

Commit:
Added data_keys field to event_descriptors

Commit:
Search on header _id field capability added. Needs debugging on userapi.commands but functional in all other apis

ericdill committed:
MNT: Updated listify and get_keys.

Still no support for multiple event_descriptors per header

ericdill committed:
ENH: Added helper function to get the calibration dictionary

ericdill committed:
ENH: Added return val for if return is nested dict

Commit:
Fixed the _id search issue related to userapi.commands.search(). Modified sample userapi code to reflect changes. All metadataStore apis now support _id (header_id) search

Commit:
Merge remote-tracking branch 'origin/devel' into devel

Commit:
Added 'tags' to run headers. There are no searches implemented yet but will be included in the next iteration

Commit:
devel branch is now master branch

Commit:
Added tags to header collection definition and updated all APIs. Tags can be inserted as a part of the headers and queried like any other parameter that is part of headers

Commit:
Fixed bulk insert issue on collectionapi.commands.record()

Commit:
Merge remote-tracking branch 'origin/master'

Conflicts:
	.gitignore
	frame_hdr_events.txt
	metadataStore/database/test.py
	metadataStore/sessionManager/databaseInit.py
	metadataStore/userapi/commands.py
	sampleCode.py
	setup.py

Commit:
Added a tablify routine "template" and modified the sample_code_userapi.py to show the changes

Commit:
Added header_versions field to header

Commit:
Sphinx documentation template added inside doc/

Commit:
Sphinx documentation template built and added to master branch

Commit:
Fixed bulk insert and search bugs in collection api. Updated documentation.

Commit:
Added doc/build

Commit:
Fixed regular expression problem on owner

Commit:
Changed documentation theme and added new fields

Commit:
Added build to documentation

Commit:
Sphinx documentation template built and added to master branch

Commit:
Changes to documentation. readthedocs.org should reflect some of these changes

Commit:
Logo added to documentation

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
Incorrect logo deleted

Commit:
Correct logo added

Commit:
Changed conf.py to create readthedocs.org code

Commit:
Changed conf.py to create readthedocs.org code

Commit:
dummy code change in collectionapi.commands for validation of formatting. To be reversed immediately

Commit:
Reversed the dummy change and added new director to path inside conf.py

Commit:
Another change to be immediately reversed inside userapi.commands

Commit:
Reversing dummy changes and modification to conf.py

Commit:
Modification to conf.py

Commit:
Reversed all dummy changes and updated the build/html

Commit:
Added __future__ to userapi commands

Commit:
Fixed minor problem within userapi

Commit:
Fixed documentation conf.py with appropriate paths

Commit:
Merge remote-tracking branch 'origin/master'

Conflicts:
	doc/build/doctrees/architecture.doctree
	doc/build/html/architecture.html
	doc/build/html/index.html
	doc/build/html/objects.inv
	doc/build/html/searchindex.js

Commit:
Changes and additions to documentation

Commit:
Addition to documentation

Commit:
pip requirements added to doc/source

Commit:
Changes and additions to documentation

Commit:
Some minor changes to sample code

Commit:
Added event qualifier to search

Commit:
Added find2 routine to metadataStore.dataapi.commands(). This closes issue/request #31.

ericdill committed:
MNT: Commented out searches that fail

Commit:
Update license

Commit:
Some minor changes commands and sample_code_raw in examples

Commit:
Merge remote-tracking branch 'origin/master'

Conflicts:
	example/sample_code_raw.py

ericdill committed:
MNT: Fixed unicode compatibility

Commit:
Merge pull request #32 from ericdill/master

MNT: Fixed unicode compatibility

Commit:
local merge

Commit:
Added data_keys to event_descriptor

ericdill committed:
MNT: Fixed collection script based on changes to event descriptor api

ericdill committed:
MNT: Fixed search

ericdill committed:
BUG: Modified metadatastore log creation

- Metadatastore will now first attempt to create a log titled:
  '/tmp/MetaDataStore.log'
- If that fails, it will append _username and create it again.

Commit:
Merge pull request #33 from ericdill/log_fixes

BUG: Modified metadatastore log creation

ericdill committed:
MNT: Removed mutables from constructor

Set optional input parameters to None and moved the defaults
into the body of __init__

ericdill committed:
MNT: Improved error messages

ericdill committed:
DOC: PEP8 conformation of docstring

Commit:
Merge pull request #34 from ericdill/input_defaults

MNT: Removed mutables from constructor

ericdill committed:
WIP: Reasonably major refactor

Commit:
Merge pull request #38 from ericdill/input_defaults

WIP: Reasonably major refactor

ericdill committed:
MNT: Implemented record with recursion

ericdill committed:
DOC: Modified frame_hdr_events based on recent changes

ericdill committed:
ENH: Added header version field

ericdill committed:
MNT: Cleaned up collection api

ericdill committed:
MNT: Cleaned up data api

- removed mutables from default parameters

ericdill committed:
MNT: MNT: Removed most defaults from database level

Logic for this commit is that you will need all of these bits of
information or the record construction will break. Therefore, the
defaults must be set at a higher level of the software

ericdill committed:
MNT: Fixes for data collection demo

ericdill committed:
MNT: Removed worthless demo & disable non-functional demos

ericdill committed:
MNT: Removed print line

Commit:
Merge pull request #39 from ericdill/recursive_record

Recursive record

tacaswell committed:
MNT : moved tests to metadataStore/tests

 - removed empty files

ericdill committed:
MNT: Added 'api' to setup.py

tacaswell committed:
DOC : add mock + numpydoc

tacaswell committed:
MNT : remove generated docs from version control

tacaswell committed:
DOC : more documentation clean up

 - added more Mocked classes
 - removed un-used html file
 - cleaned up rst markup
 - don't blindly catch exceptions during DB connection

tacaswell committed:
DOC : reformatted and corrected insert_event docs

tacaswell committed:
MNT/DOC : factored and simplified create_event

 - moved key existence validation to helper function
 - made documentation match code
 - removed un-used imports

Commit:
Merge pull request #40 from tacaswell/mnt_organization

Mnt organization [WIP]

Commit:
Revert "Mnt organization [WIP]"

Commit:
Merge pull request #41 from NSLS-II/revert-40-mnt_organization

Revert "Mnt organization [WIP]"

tacaswell committed:
Revert "Revert "Mnt organization [WIP]""

tacaswell committed:
Merge pull request #42 from NSLS-II/revert-41-revert-40-mnt_organization

Revert "Revert "Mnt organization [WIP]""

tacaswell committed:
DOC : tweak installs for rtd

tacaswell committed:
Merge pull request #43 from tacaswell/rtd_fix

DOC : tweak installs for rtd

Commit:
Merge branch 'master' of https://github.com/NSLS-II/metadataStore

Commit:
Fixed log file createion failure when permission is denied. Closes #45

Commit:
Delete debug print statement

Commit:
data_keys '.' replaced with [dot] in event_descriptor

Commit:
Mechanism to ensure event['data'].keys() is valid with respect to event_descriptor['data_keys']

tacaswell committed:
BLD : add conda recipe

Commit:
data_keys can have '.' in the name. The engine converts '.' to [dot]

Commit:
Fixed sample_code_raw.

Commit:
Fixed minor issue on find() replacing [dot] with '.'

Commit:
Added dot replacement mechanism to find(scan_id='last')

Commit:
Fixed find(scan_id='last') mechanism dot replacement issue

Commit:
Update README.md

ericdill committed:
ENH: Added api/analysis.py

MNT: Import find_last from dataapi/commands

Commit:
Added find_last() to dataapi. Returns a tuple composed of most recently created run_header and its associated info

Commit:
Added find_last() to dataapi. Returns a tuple composed of most recently created run_header and its associated info

Commit:
Added find_last() collection api namespace

Commit:
Fixed sample_collection_code.py after addition of data_key validation for events and event_descriptor

Commit:
Fix sample_collection_code.py after data_key validation added to event_descriptors and events

Commit:
Fixed find_last event return issue.

Commit:
Fixed find_last event return issue.

Commit:
Reflected the changes to find(scan_id=last

Commit:
Added both find examples to collection code sample

Commit:
Fixed find(scan_id=s_id, data=True) issue

Commit:
Fixed find(scan_id=s_id, data=True) issue

Commit:
[dot] inside event data keys replaced with [dot] in find(scan_id=some_id, data=True)

Commit:
time format on search changed from datetime to unix time

Commit:
Search for find(scan_id=current) made obsolete. Lingo should be revisited and determined

ericdill committed:
Merge remote-tracking branch 'caswell/conda' into analysis_api

Commit:
Merge pull request #50 from ericdill/analysis_api

ENH: Added api/analysis.py

Commit:
Fixed __decode_cursor(event_cursor) in case we need to re-utilize. Will use it to clean up the code

Commit:
Delete frame_hdr_events_proposed.txt

Commit:
Update pip-req.txt

Commit:
Update conf.py

Commit:
Hardcoded config info for CSX into sessionManager/databaseInit.py

ericdill committed:
MNT: added print statements

ericdill committed:
Merge remote-tracking branch 'upstream/master'

ericdill committed:
MNT: Revert hard coding mds location

Closes #52

Commit:
WIP: API refactoring. Inner facing metadataStore.dataapi.commands is now under mdstools

Commit:
WIP: Changes to setup.py in compliance with refactoring and minor fixes(deleted debug print statement on header cursor etc)

Commit:
WIP: Added create_event, validate_dict_keys from former collectionapi.commands in order to expose event creation via metadataStore.api.collection

Commit:
WIP: Changes to config files. Until we find a better way that most agrees upon, dataBroker.conf is read from user's home directory in order to avoid confusion.

Commit:
WIP: Added unit test templates and deleted some irrelevant files and directories

Commit:
ENH: Header custom field content validity check added

ericdill committed:
slight improvement to type validation on custom header field

ericdill committed:
MNT: Update example and slight change to init

ericdill committed:
ENH: Validate first and second layer of keys in custom field

ericdill committed:
MNT: Recursive validation of custom field

ericdill committed:
MNT: Recursively validate the entire header

ericdill committed:
MNT: Added tuple to list of valid types

ericdill committed:
MNT: Removed print statement

Commit:
Merge pull request #54 from ericdill/development

Development

Commit:
Merge pull request #55 from NSLS-II/development

Development

ericdill committed:
DOC: Updated schema docs to prototype frame store

ericdill committed:
DOC: Further updates to schema documentation

ericdill committed:
DOC: Spelling is hard

danielballan commented on frame_hdr_events.txt:
```diff
@@ -37,13 +38,25 @@ The return value will be formatted like this:
+            external_data_keys: {
+                ext_pv1,
+                ...
+                ext_pvM
```
I recall we said that `data_keys` would be all-inclusive, so for illustrative purposes the list of `data_keys` should also include `ext_pv1`, `ext_pv2`, etc. "Internal" data keys can be computed as needed using set logic.

ericdill committed:
DOC: Proposed schema update re: Wilkins and Chabot

Updated the proposed schema outline based on separate conversations
with @stuwilkins and @dchabot
